packages/iris-grid/src/sidebar/CustomColumnBuilder.tsx

```diff
@@ -334,21 +336,30 @@ class CustomColumnBuilder extends Component<
   renderSaveButton(): ReactElement {
     const { inputs, isCustomColumnApplying, isSuccessShowing } = this.state;
     const saveText = inputs.length > 1 ? 'Save Columns' : 'Save Column';
+    const areNamesValid = inputs.every(({ name }) =>
+      DbNameValidator.isValidColumnName(name)
+    );
+    const areNamesUnique =
+      new Set(inputs.map(({ name }) => name)).size === inputs.length;
 
     return (
       <Button
         kind={isSuccessShowing ? 'success' : 'primary'}
         className={classNames('btn-apply', {
           'btn-spinner': isCustomColumnApplying,
         })}
-        disabled={isSuccessShowing || isCustomColumnApplying}
+        disabled={
```

Comment:
This is disabling the button when there are no custom columns; if you add a custom column, then want to remove it, there's no long a way to do that. May want to consider just dropping any that are both blank name and formula.

Also weird behaviour if you add a name, but then don't add a formula - you can still hit Save, but the column with the name doesn't appear. May want to check there are no blank formulas.

Reply:
The previous behavior was to drop anything that had either part blank.

But I also just discovered if both are blank and you save, it gets stuck in an infinite loader. Checked on Jackson too and same behavior. So need to check/filter that if allowing save w/ all blank

Reply:
I made the button visible if there's no custom columns. I also fixed the loader issue in the case where you had no custom columns and clicked save with no custom columns. That case triggered the loading state, but the grid panel listens for a table change event to stop the loader. That event never happens. The column builder panel also listens for the same event to dismiss the loader

Reply:
Adding a custom column with a name and formula, then removing the formula and hitting save still succeeds and leaves the previous formula on the table

Reply:
Looks like that bug exists in the published version too. I'll fix it

Reply:
I tried adding a test for this w/ RTL, but it seems if I load the panel w/ an existing formula, I can't delete the text in monaco. Oddly, I can add w/ keyboard events (adds to the front of the existing string), but delete key doesn't delete the existing text, select all then delete doesn't either

Reply:
Managed to get a test working by creating a new custom column, saving, then deleting the text and saving again. Guess it's something to do with monaco starting w/ the text vs adding it via RTL keyboard events.